### ISSUE TITLE
New Shop Unable To Change Address In Shop Panel

### ISSUE THREAD

**Issue by willmoss1000** [reactioncommerce/reaction#3836]
## Prerequisites
* [y] Are you running the latest version? **1.8.1**
* [y] Are you able to consistently reproduce the issue?
* [y] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues) - tried to, but couldn't find anything. Also posted in gitter to check.

## Issue Description
Upon signing up as a new seller in marketplace, (by clicking on 'become seller', or via invitation), going to the 'Shop' tab on the RHS and trying to input an address results in the following toast error showing, and no submission: "Shop general settings update failed. Error: Access denied [403]"

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Fresh init of reaction commerce
2. Turn on marketplace, and allow seller signup
3. Sign up a new account.
4. Go to 'Shop' tab on RHS and try to input the address.
5. Error shows.

## Possible Solution
Did not exist for me in 1.8.1. Looking in the console suggests that the new seller does not have permission to change their address in the 'Shop' panel.

### Versions
Node: 8.9.0
NPM: 5.5.1
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.28.0
Reaction: 1.8.1
Reaction branch: master

**Comment by zenweasel** [comment]
Verified this is happening in the current `1.8.1` branch. Bumped up to `impact-critical` as you can't create a shop with this bug

**Comment by pmn4** [comment]
I dug into this a bit and it seems like when you create a shop and shop owner, the shop owner's `activeShopId` user preference is not set. Then, when you try to update, the code hits this block:

``` JavaScript
  Security.defineMethod("ifHasRoleForActiveShop", {
    fetch: [],
    transform: null,
    allow(type, arg, userId) {
      if (!arg) throw new Error("ifHasRole security rule method requires an argument");
      if (arg.role) {
        // Note: userId is passed to getShopId to ensure that it returns the correct shop based on the User Preference
        // if not passed, getShopId can default to primaryShopId if Meteor.userId is not available in the context the code is run
        return Roles.userIsInRole(userId, arg.role, Reaction.getShopId(userId));
      }
      return Roles.userIsInRole(userId, arg);
    }
  });
```
[source](https://github.com/reactioncommerce/reaction/blob/master/server/startup/collection-security.js#L43-L55)

Specifically, this line:
``` JavaScript
return Roles.userIsInRole(userId, arg.role, Reaction.getShopId(userId));
```

Since the user's preference is not set, `getShopId` defaults to finding the shop on the current domain, which probably means the primary shop.

It's a little odd that you have to look up the user preference when it is clearly known and could be passed to this method (in fact, it _is_! as an unnamed parameter).

I dug a little further, and when a shop owner is invited, the activeShopId is set in the wrong place:
It is written to (Meteor's) `users.profile.preferences.reaction.activeShopId`
it is read from (Reaction's) `Accounts.profile.preferences.reaction.activeShopId`
(I assume this was a refactor in a recent version?)

Additionally, when a user creates their own shop, `activeShopId` is not set at all

IMO, there are two things that should happen: 1) save the activeShopId to the correct collection in all places where a Shop Owner can create their account; 2) don't rely on this user preference when saving shop data, instead, get the shop id from the request

**Comment by pmn4** [comment]
Lastly, instead of writing directly, the `setUserPreferences` method should be used [here](https://github.com/reactioncommerce/reaction/blob/master/server/methods/accounts/accounts.js#L624)

**Comment by zenweasel** [comment]
@pmn4 I don't think we want to get the shopId from the client for security reasons

**Comment by zenweasel** [comment]
Closed via #3845
